### ISSUE TITLE
Missing type `deliveredAmount` for `types.d.ts` for getTransaction API Call

### ISSUE THREAD

**Issue by Daniel-MCO** [ripple/ripple-lib#995]
According to the `getTransaction` api call from https://developers.ripple.com/rippleapi-reference.html#gettransaction, the `deliveredAmount` field is returned from the response.

However, when referring to the `types.d.ts` from the ripple-lib, it appears that the type `Outcome` does not contain this field as even an optional.

Can this be updated back into the `types.d.ts`, and/or if anyone know whether I am looking at the wrong place for this typing?

For the change, I would envision something like the following:

```
export declare type Outcome = {
    result: string;
    ledgerVersion: number;
    indexInLedger: number;
    fee: string;
    balanceChanges: {
        [key: string]: [{
            currency: string;
            counterparty?: string;
            value: string;
        }];
    };
    orderbookChanges: Object;
    deliveredAmount?: {
      currency: string;
      value: string;
      counterparty: string;
    };
    timestamp?: string;
};
```

Thanks.

**Comment by intelliot** [comment]
Good catch! I think the place to add it is here: https://github.com/ripple/ripple-lib/blob/develop/src/transaction/types.ts#L72

There are a couple minor tweaks I would make to what you're envisioning:
- orderbookChanges can be `object` with lowercase o
- deliveredAmount.counterparty can be optional

What do you think - would you like to open a PR?

**Comment by Daniel-MCO** [comment]
Hey Elliot, thanks for the fast feedback! Yeah, I will make a fork and make a PR with the branch `add-delivered-amount-type` including the suggestions you've made.

Hope this is the right way to make a PR for `ripple-lib`.

**Comment by intelliot** [comment]
Released in 1.2.3
